### ISSUE TITLE
Index out of range

### ISSUE THREAD

**Issue by BJReplay** [BJReplay/ha-solcast-solar#209]
A "index out of range" issue:

```
2024-10-27 23:15:00.065 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] Exception in __get_forecast_pv_moment(): list index out of range: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1935, in __get_forecast_pv_moment
    res = self.__get_moment(site, _data_field, (time_utc - day_start).total_seconds())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1814, in __get_moment
    return self._forecasts_moment['all' if site is None else site][self._use_data_field if _data_field is None else _data_field][int(n_min / 300)]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
IndexError: list index out of range

2024-10-27 23:15:00.067 ERROR (MainThread) [custom_components.solcast_solar.sensor] Unable to get sensor value: list index out of range: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/sensor.py", line 441, in _handle_coordinator_update
    self._sensor_data = self._coordinator.get_sensor_value(self.entity_description.key)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/coordinator.py", line 347, in get_sensor_value
    return self.solcast.get_power_n_mins(60)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1508, in get_power_n_mins
    return round(1000 * self.__get_forecast_pv_moment(time_utc, site=site, _use_data_field=_use_data_field))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1935, in __get_forecast_pv_moment
    res = self.__get_moment(site, _data_field, (time_utc - day_start).total_seconds())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1814, in __get_moment
    return self._forecasts_moment['all' if site is None else site][self._use_data_field if _data_field is None else _data_field][int(n_min / 300)]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
IndexError: list index out of range

2024-10-27 23:15:00.067 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] Exception in __get_forecast_pv_moment(): list index out of range: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1935, in __get_forecast_pv_moment
    res = self.__get_moment(site, _data_field, (time_utc - day_start).total_seconds())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1814, in __get_moment
    return self._forecasts_moment['all' if site is None else site][self._use_data_field if _data_field is None else _data_field][int(n_min / 300)]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
IndexError: list index out of range

2024-10-27 23:15:00.068 ERROR (MainThread) [custom_components.solcast_solar.sensor] Unable to get sensor value: list index out of range: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/sensor.py", line 405, in extra_state_attributes
    return self._coordinator.get_sensor_extra_attributes(self.entity_description.key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/coordinator.py", line 417, in get_sensor_extra_attributes
    return self.solcast.get_sites_power_n_mins(60)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1522, in get_sites_power_n_mins
    res[site['resource_id']] = self.get_power_n_mins(n_mins, site=site['resource_id'])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1508, in get_power_n_mins
    return round(1000 * self.__get_forecast_pv_moment(time_utc, site=site, _use_data_field=_use_data_field))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1935, in __get_forecast_pv_moment
    res = self.__get_moment(site, _data_field, (time_utc - day_start).total_seconds())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/solcastapi.py", line 1814, in __get_moment
    return self._forecasts_moment['all' if site is None else site][self._use_data_field if _data_field is None else _data_field][int(n_min / 300)]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
IndexError: list index out of range
```

_Originally posted by @isorin in https://github.com/BJReplay/ha-solcast-solar/discussions/202#discussioncomment-11085948_

**Comment by Nilogax** [comment]
Just checked my logs and have the same issue at the same time

```
This error originated from a custom integration.

Logger: custom_components.solcast_solar.solcastapi
Source: custom_components/solcast_solar/solcastapi.py:1943
integration: Solcast PV Forecast (documentation, issues)
First occurred: 27 October 2024 at 23:15:00 (24 occurrences)
Last logged: 27 October 2024 at 23:55:00

Exception in __get_forecast_pv_moment(): list index out of range: Traceback (most recent call last): File "/config/custom_components/solcast_solar/solcastapi.py", line 1935, in __get_forecast_pv_moment res = self.__get_moment(site, _data_field, (time_utc - day_start).total_seconds()) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/solcast_solar/solcastapi.py", line 1814, in __get_moment return self._forecasts_moment['all' if site is None else site][self._use_data_field if _data_field is None else _data_field][int(n_min / 300)] ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^ IndexError: list index out of range
```

**Comment by autoSteve** [comment]
Solved by v4.2.5, pending. Adding this to the "will be closed by" list.

I have hit this well beyond anyone. @Nilogax, @isorin, @BJReplay.
